renderdoc/driver/gl/gl_pixelhistory.cpp

```diff
@@ -38,22 +38,29 @@ bool isDirectWrite(ResourceUsage usage)
           usage == ResourceUsage::GenMips);
 }
 
+struct CopyFramebuffer
+{
+  GLuint framebufferId;
+  GLuint colorTextureId;
+  GLuint dsTextureId;
+  GLuint depthTextureId;
+  GLuint stencilTextureId;
+  GLuint stencilViewId;
+};
+
 struct GLPixelHistoryResources
 {
   // Used for offscreen rendering for draw call events.
-  GLuint colorImage;
-  GLuint dsImage = 0;
-  GLuint frameBuffer;
   GLuint fullPrecisionColorImage;
   GLuint fullPrecisionDsImage;
   GLuint fullPrecisionFrameBuffer;
   GLuint primitiveIdFragmentShader;
   GLuint primitiveIdFragmentShaderSPIRV;
+  GLuint msCopyComputeProgram;
+  GLuint msCopyDstBuffer;
+  GLuint msCopyUniformBlockBuffer;
   std::unordered_map<GLuint, GLuint> programs;
-  bool depthTextureAttachedToFrameBuffer;
-  bool stencilTextureAttachedToFrameBuffer;
-  GLuint depthImage = 0;
-  GLuint stencilImage = 0;
+  std::map<std::tuple<GLenum, GLenum, uint32_t>, CopyFramebuffer> copyFramebuffers;
```

Comment:
Please don't use `std::tuple`. If you need a complex key here, prefer making a separate named struct and use that.